app/models/user.py

```diff
@@ -212,6 +212,8 @@ def platform_admin(self):
         return self._platform_admin and not session.get("disable_platform_admin_view", False)
 
     def has_permissions(self, *permissions, restrict_admin_usage=False, allow_org_user=False):
+        from app.models.service import Service
```

Comment:
Think you can do
```suggestion
        from app import current_service
```

…then we are only hitting the API/Redis once per request context

Reply:
good shout. needs me to fix a couple of tests but worth doing